virtual/environment.go

```diff
@@ -530,7 +531,13 @@ func (r *environment) InvokeActorDirectStream(
 			heartbeatResult.ServerVersion, serverVersion)
 	}
 
-	return r.activations.invoke(ctx, reference, operation, create.InstantiatePayload, payload, false)
+	// Wrap instantiation payload into a struct that provides metadata to the actor
+	b, err := json.Marshal(types.InstantiatePayload{Payload: create.InstantiatePayload, IsWorker: false})
```

Comment:
I think we should push these down into `activations.go` that way we can avoid the allocations/marshalling work except for the case where a new actor is actually being created which will make it ~free